src/OrbitQt/HistogramWidget.cpp

```diff
@@ -92,75 +101,64 @@ static void DrawVerticalLine(QPainter& painter, const QPoint& start, int length)
   painter.drawLine(start, {start.x(), start.y() + length});
 }
 
-namespace {
-struct Ticks {
-  std::vector<QString> labels;
-  std::vector<double> values;
-  int precision{};
-};
-}  // namespace
+[[nodiscard]] static uint32_t TickCount(double min, double max, double step) {
+  double first = std::ceil(min / step) * step;
+  if (first > max) return 0;
+  return static_cast<uint32_t>(std::floor((max - first) / step)) + 1;
+}
 
-template <typename Rounder>
-[[nodiscard]] static double Round(double x, int precision, Rounder&& rounder) {
-  const double power_of_ten = pow(10.0, precision);
-  return rounder(x * power_of_ten) / power_of_ten;
-}
-
-[[nodiscard]] static std::vector<double> RoundDoubles(const std::vector<double>& values,
-                                                      int precision) {
-  std::vector<double> rounded;
-  for (size_t i = 0; i < values.size(); ++i) {
-    double result = 0;
-    if (i == 0) {
-      result = Round(values[i], precision, static_cast<double (*)(double)>(std::ceil));
-    } else if (i + 1 == values.size()) {
-      result = Round(values[i], precision, static_cast<double (*)(double)>(std::floor));
-    } else {
-      result = Round(values[i], precision, static_cast<double (*)(double)>(std::round));
-    }
-    rounded.push_back(result);
+[[nodiscard]] static std::vector<double> MakeLabelValues(double min, double max, double step) {
+  double current = std::ceil(min / step) * step;
+  std::vector<double> result = {current};
+
+  while (current <= max) {
+    result.push_back(current);
+    current += step;
   }
-  return rounded;
-}
 
-[[nodiscard]] static Ticks MakeTicks(const std::vector<double>& values, int precision) {
-  std::vector<QString> labels;
-  const std::vector<double> rounded_values = RoundDoubles(values, precision);
-  std::transform(
-      std::begin(rounded_values), std::end(rounded_values), std::back_inserter(labels),
-      [precision](const double value) { return QString::number(value, 'f', precision); });
-  return {labels, rounded_values, precision};
+  return result;
 }
 
-template <typename T>
-static bool AreAllUnique(std::vector<T> vector) {
-  std::sort(vector.begin(), vector.end());
-  return std::unique(vector.begin(), vector.end()) == vector.end();
-}
+[[nodiscard]] static TickStep ChooseBestStep(double min, double max,
```

Comment:
Nit: A comment to explain this method? (for example to explain that we are not always going to get the desired number of ticks)